src/WebJobs.Script.Scaling/AzureScaleHandler.cs

```diff
@@ -0,0 +1,287 @@
+                    {
+                        return false;
+                    }
+
+                    throw;
+                }
+            }
+        }
+
+        /// <summary>
+        /// This requests FE to remove this specific worker.
+        /// - FE should return success regardless if worker belongs.
+        /// - Any unexpected error will throw.
+        /// </summary>
+        public async Task RemoveWorker(string activityId, IWorkerInfo worker)
+        {
+            var stampHostName = GetStampHostName(worker.StampName);
+            var details = string.Format("remove worker request from {0}:{1}", AzureEnvironments.CurrentStampName, AzureEnvironments.WorkerName);
+            var pathAndQuery = string.Format("https://{0}/operations/removeworker/{1}/{2}?token={3}",
+                stampHostName,
+                AzureEnvironments.SiteName,
+                worker.WorkerName,
+                GetToken());
+
+            using (var response = await SendAsync(activityId, HttpMethod.Delete, pathAndQuery, worker, details))
+            {
+                response.EnsureSuccessStatusCode();
+            }
+        }
+
+        public static string GetStampHostName(string stampName)
+        {
+            return string.Format("{0}.cloudapp.net", stampName);
+        }
+
+        public static string GetToken()
+        {
+            if (string.IsNullOrEmpty(_token) || _tokenExpiredUtc < DateTime.UtcNow)
+            {
+                var expiredUtc = DateTime.UtcNow.Add(TokenValidity);
+                var token = ScaleUtils.GetToken(expiredUtc);
+                _token = WebUtility.UrlEncode(token);
+                _tokenExpiredUtc = expiredUtc;
+            }
+
+            return _token;
+        }
+
+        private static async Task<HttpResponseMessage> SendAsync(string activityId, HttpMethod method, string pathAndQuery, IWorkerInfo workerInfo, string details)
+        {
+            using (var client = CreateHttpClient(activityId, workerInfo, details))
+            {
+                return await client.SendAsync(new HttpRequestMessage(method, pathAndQuery));
+            }
+        }
+
+        [SuppressMessage("Microsoft.Reliability", "CA2000:Dispose objects before losing scope", Justification = "By design")]
+        private static HttpClient CreateHttpClient(string activityId, IWorkerInfo workerInfo, string details)
+        {
+            var handler = HttpMessageHandler;
+            if (handler == null)
+            {
+                var webHandler = new WebRequestHandler();
+                if (!AzureEnvironments.ValidateCertificates.Value)
+                {
+                    webHandler.ServerCertificateValidationCallback = ServerCertificateValidation;
+                }
+
+                handler = webHandler;
+            }
+
+            var client = new HttpClient(new HttpLoggingHandler(handler, activityId, workerInfo, details));
```

Comment:
Can we use a single cached HttpClient for the lifetime of the app, rather than new one up each time? That is the general performance guideline for HttpClient. https://stackoverflow.com/questions/22560971/what-is-the-overhead-of-creating-a-new-httpclient-per-call-in-a-webapi-client